src/test/kotlin/it/pagopa/wallet/services/WalletServiceTest.kt

```diff
@@ -2072,6 +2074,75 @@ class WalletServiceTest {
+                walletClientInfo["IO"] = WalletClientDto().status(WalletClientStatusDto.ENABLED)
+                val walletInfoDto =
+                    WalletInfoDto()
+                        .walletId(UUID.fromString(wallet.id))
+                        .status(WalletStatusDto.valueOf(wallet.status))
+                        .paymentMethodId(wallet.paymentMethodId)
+                        .userId(wallet.userId)
+                        .updateDate(OffsetDateTime.parse(wallet.updateDate.toString()))
+                        .creationDate(OffsetDateTime.parse(wallet.creationDate.toString()))
+                        .applications(
+                            wallet.applications.map { application ->
+                                WalletApplicationInfoDto()
+                                    .name(application.id)
+                                    .status(WalletApplicationStatusDto.valueOf(application.status))
+                            }
+                        )
+                        .details(
+                            WalletCardDetailsDto()
+                                .type((wallet.details as CardDetails).type)
+                                .brand((wallet.details as CardDetails).brand)
+                                .expiryDate((wallet.details as CardDetails).expiryDate)
+                                .lastFourDigits((wallet.details as CardDetails).lastFourDigits)
+                        )
+                        .clients(walletClientInfo)
+
+                given {
+                        walletRepository.findByIdAndUserId(
+                            eq(WALLET_UUID.value.toString()),
+                            eq(USER_ID.id.toString())
+                        )
+                    }
+                    .willAnswer { Mono.just(wallet) }
+
+                /* test */
+
+                StepVerifier.create(walletService.findWallet(WALLET_UUID.value, USER_ID.id))
+                    .expectNext(walletInfoDto)
+                    .verifyComplete()
+            }
+        }
+    }
+
+    @Test
+    fun `should find wallet document with client info about last usage`() {
+        /* preconditions */
+
+        mockStatic(UUID::class.java, Mockito.CALLS_REAL_METHODS).use {
+            it.`when`<UUID> { UUID.randomUUID() }.thenReturn(mockedUUID)
+
+            mockStatic(Instant::class.java, Mockito.CALLS_REAL_METHODS).use {
+                print("Mocked instant: $mockedInstant")
+                it.`when`<Instant> { Instant.now() }.thenReturn(mockedInstant)
+
+                val wallet = walletDocumentStatusValidatedCard(BRAND, TEST_FULL_INFO_CLIENTS)
+
+                val walletClientInfo = HashMap<String, WalletClientDto>()
+                walletClientInfo["IO"] =
+                    WalletClientDto()
+                        .status(WalletClientStatusDto.ENABLED)
+                        .lastUsage(OffsetDateTime.parse(wallet.clients["IO"]?.lastUsage))
```

Comment:
If this must be present I'd drop `?` and use `!!` so that we get an error instead if it's `null`
```suggestion
                        .lastUsage(OffsetDateTime.parse(wallet.clients["IO"]!!.lastUsage))
```